pkg/tests/api/alerting/api_notification_channel_test.go

```diff
@@ -0,0 +1,963 @@
+		}`,
+	},
+}
+
+func getAlertmanagerConfig(channelAddr string) string {
+	return strings.ReplaceAll(alertmanagerConfig, "CHANNEL_ADDR", channelAddr)
+}
+
+func getExpAlertmanagerConfigFromAPI(channelAddr string) string {
+	return strings.ReplaceAll(expAlertmanagerConfigFromAPI, "CHANNEL_ADDR", channelAddr)
+}
+
+// alertNames are name of alerts to be sent. This should be in sync with
+// the routes that we define in Alertmanager config.
+// EmailAlert and TelegramAlert are missing because they don't
+// send a JSON. Email and POST body are yet to be supported in the tests.
+var alertNames = []string{"DingDingAlert", "SlackAlert1", "SlackAlert2", "PagerdutyAlert", "TeamsAlert", "WebhookAlert"}
+
+func getRulesConfig(t *testing.T) string {
+	interval, err := model.ParseDuration("10s")
+	require.NoError(t, err)
+	rules := apimodels.PostableRuleGroupConfig{
+		Name:     "arulegroup",
+		Interval: interval,
+	}
+
+	// Create rules that will fire as quickly as possible for all the routes.
+	for _, alertName := range alertNames {
+		rules.Rules = append(rules.Rules, apimodels.PostableExtendedRuleNode{
+			GrafanaManagedAlert: &apimodels.PostableGrafanaRule{
+				Title:     alertName,
+				Condition: "A",
+				Data: []ngmodels.AlertQuery{
+					{
+						RefID: "A",
+						RelativeTimeRange: ngmodels.RelativeTimeRange{
+							From: ngmodels.Duration(time.Duration(5) * time.Hour),
+							To:   ngmodels.Duration(time.Duration(3) * time.Hour),
+						},
+						DatasourceUID: "-100",
+						Model: json.RawMessage(`{
+							"type": "math",
+							"expression": "2 + 3 > 1"
+						}`),
+					},
+				},
+			},
+		})
+	}
+
+	b, err := json.Marshal(rules)
+	require.NoError(t, err)
+
+	return string(b)
+}
+
+func TestNotificationChannels(t *testing.T) {
+	dir, path := testinfra.CreateGrafDir(t, testinfra.GrafanaOpts{
+		EnableFeatureToggles: []string{"ngalert"},
+		DisableAnonymous:     true,
+	})
+
+	store := testinfra.SetUpDatabase(t, dir)
+	store.Bus = bus.GetBus()
+	grafanaListedAddr := testinfra.StartGrafana(t, dir, path, store)
+
+	mockChannel := newMockNotificationChannel(t, grafanaListedAddr)
+	amConfig := getAlertmanagerConfig(mockChannel.server.Addr)
+	rulesConfig := getRulesConfig(t)
+
+	// Overriding some URLs to send to the mock channel.
+	channels.SlackAPIEndpoint = fmt.Sprintf("http://%s/slack_recvX/slack_testX", mockChannel.server.Addr)
+	channels.PagerdutyEventAPIURL = fmt.Sprintf("http://%s/pagerduty_recvX/pagerduty_testX", mockChannel.server.Addr)
+	channels.TelegramAPIURL = fmt.Sprintf("http://%s/telegram_recv/bot%%s", mockChannel.server.Addr)
+
+	// Create a user to make authenticated requests
+	require.NoError(t, createUser(t, store, models.ROLE_EDITOR, "grafana", "password"))
+
+	{
+		// There are no notification channel config initially.
+		alertsURL := fmt.Sprintf("http://grafana:password@%s/api/alertmanager/grafana/config/api/v1/alerts", grafanaListedAddr)
```

Comment:
Is this right? I would expect the default configuration to be present...

Reply:
It passes, gives 404, so something wrong?

Reply:
This is meant to be, its my PR of #33539 that modifies this behaviour.